lib/resty/acme/autossl.lua

```diff
@@ -520,6 +542,25 @@ function AUTOSSL.is_domain_whitelisted(domain, is_new_cert_needed)
   end
 end
 
+local function find_wildcard_domain(domain)
```

Comment:
i would suggest to not add the wildcard matcher here by default, but let user to use `__index` meta method to match it if they wish, as iterrateing the table every time would be not performant.
An alternative way is to compile all wildcard domains into a single regular expression string, (ref: https://github.com/Kong/kong/blob/master/kong/plugins/acme/handler.lua#L33)